### PR TITLE
Refactor code PR

### PR DESCRIPTION
**Pull request by adamteddychang**
In this Pull request we: 
- Made sure our code obeys the rules of YAGNI, KISS, DRY, and Ruby style guide
- Followed SOLID principles
- Refactored `main.rb` so it only has the necessary starter method in its class
- This project is done in collaboration with @xtrahuman

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    puts
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
```
Comment (mikethreels) [inline app.rb]: - [x] this is exactly the same code as you are using in the main.rb file which is not in line with the DRY rules, please make sure that you don't write pieces of code twice unnecessarily.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age, specialization, name)
+    @people << teacher
+
+    puts 'Teacher created successfully'
+    sleep 0.75
+  end
```
Comment (mikethreels) [inline app.rb]: - [x] in the `create_teacher` & `create_student` methods you're writing the same code twice, which is not in line with the DRY rules, please make sure that you don't write pieces of code twice unnecessarily. you could consider joining this code in the `create_person` method
- [x] in addition to that in these methods you are printing to the console, getting information and saving the data which is not in line with the single responsibility rule, please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+    print 'Author: '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+
+    puts 'Book added successfully'
+    sleep 0.75
+  end
```
Comment (mikethreels) [inline app.rb]: - [x]  in these methods you are printing to the console, getting information and saving the data which is not in line with the single responsibility rule, please have a look at this article for more information.

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+  print 'Date: '
+  date = gets.chomp.to_s
+
+  rental = Rental.new(date, @books[book_id], @people[person_id])
+  @rentals << rental
+
+  puts 'Rental created successfully'
+  sleep 0.75
+end
```
Comment (mikethreels) [inline app.rb]: - [x]  in these methods you are printing to the console, getting information and saving the data which is not in line with the single responsibility rule, please have a look at this article for more information.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._
- [ ] Because your readme has not changed in this PR I'm not able to leave an inline comment, therefore I'll leave a comment here. please make sure to explain how to get the project on your local repo as if you are explaining it to a novice, include how to clone and go into the folder and how to run the application at least.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omar-labana** [comment]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**